tests/stdlib/tasynchttpserver_transferencoding.nim

```diff
@@ -1,5 +1,6 @@
 discard """
   matrix: "--gc:arc --threads:on; --gc:arc --threads:on -d:danger; --threads:on"
+  disabled: "freebsd"
```

Comment:
always disable the minimum needed to make things work (to avoid future additional regressions)

based on https://github.com/nim-lang/Nim/issues/18120#issuecomment-850753742 (thanks for doing this test!), could you try something like:

```
discard """
  matrix: "--gc:arc --threads:on -d:nimD20210528T201138; --gc:arc --threads:on -d:danger; --threads:on"
"""

when defined(freebsd) and defined(nimD20210528T201138):
  # xxx investigate in future work
  discard
else:
  # rest of test
```

Reply:
So, I've been trying this setup for the past few minutes, and even with literally the entire file in the `#rest of test` section (including all of the imports), this fails the same.

```nim
discard """
  matrix: "--gc:arc --threads:on -d:nimD20210528T201138; --gc:arc --threads:on -d:danger; --threads:on"
"""

when defined(freebsd) and defined(nimD20210528T201138):
  # xxx investigate in future work
  discard
else:
  import httpclient, asynchttpserver, asyncdispatch, asyncfutures
  import net

  import std/asyncnet
  import std/nativesockets

  const postBegin = """
POST / HTTP/1.1
Transfer-Encoding:chunked

"""

  template genTest(input, expected: string) =
    proc handler(request: Request, future: Future[bool]) {.async, gcsafe.} =
      doAssert(request.body == expected)
      doAssert(request.headers.hasKey("Transfer-Encoding"))
      doAssert(not request.headers.hasKey("Content-Length"))
      future.complete(true)
  ...
```

Output:
```
root@freebsd-s-1vcpu-1gb-tor1-01:~/Nim # ./koch test run tests/stdlib/tasynchttpserver_transferencoding.nim
bin/nim cc --opt:speed testament/testament
Hint: used config file '/root/Nim/config/nim.cfg' [Conf]
Hint: used config file '/root/Nim/config/config.nims' [Conf]
Hint: used config file '/root/Nim/testament/testament.nim.cfg' [Conf]
...................................................................................
/root/Nim/testament/categories.nim(477, 5) Hint: 'readOnly' is declared but not used [XDeclaredButNotUsed]
/root/Nim/testament/categories.nim(476, 5) Hint: 'writeOnly' is declared but not used [XDeclaredButNotUsed]
/root/Nim/testament/categories.nim(481, 12) Hint: 'test' is declared but not used [XDeclaredButNotUsed]
Hint:  [Link]
Hint: 94961 lines; 3.843s; 181.281MiB peakmem; proj: /root/Nim/testament/testament; out: /root/Nim/testament/testament [SuccessX]
Hint: ***SLOW, DEBUG BUILD***; -d:release makes code run faster. [BuildMode]
/root/Nim/testament/testament --nim:bin/nim run tests/stdlib/tasynchttpserver_transferencoding.nim
FAIL: tests/stdlib/tasynchttpserver_transferencoding.nim c         ( 0.79 sec)
Test "tests/stdlib/tasynchttpserver_transferencoding.nim" in category "stdlib"
Failure: reExitcodesDiffer
Expected:
exitcode: 0

Gotten:
exitcode: 1

Output:


diff --git a/tmp/diffStrings_a_iKMEIPtJ b/tmp/diffStrings_b_KeoMBkOZ
index 031aaaad3..88f2ffa0c 100644
--- a/tmp/diffStrings_a_iKMEIPtJ
+++ b/tmp/diffStrings_b_KeoMBkOZ
@@ -1 +1,3 @@
-exitcode: 0
\ No newline at end of file
+exitcode: 1
+
+Output:

PASS: tests/stdlib/tasynchttpserver_transferencoding.nim c         (10.89 sec)
PASS: tests/stdlib/tasynchttpserver_transferencoding.nim c         ( 4.73 sec)
FAILURE! total: 3 passed: 2 skipped: 0 failed: 1
tests failed
```

Any ideas? :/

Reply:
3 options:
* add `tests/stdlib/tasynchttpserver_transferencoding.config.nims` containing something like:
```nim
when defined(freebsd):
  # xxx investigate
  switch("define", "danger")
```
* test it via trunner.nim which gives you more control than testament
* or simply as you did in your PR via `disabled: "freebsd"` (which then doesn't test on freebsd at all)

either way is good enough for now